web/beacon-app/src/features/onboarding/shared/utils.ts

```diff
@@ -31,27 +35,68 @@ export const getProfessionOptions = () => [
 ];
 
 export const getCurrentStepFromMember = (member: any) => {
-  let current = 0;
+  let current = 1;
   const hasOrganization = member?.organization?.length > 0;
   const hasWorkspace = member?.workspace?.length > 0;
   const hasName = member?.name?.length > 0;
   const hasProfessionSegment = member?.profession_segment?.length > 0;
+  const hasDeveloperSegment = member?.developer_segment?.length > 0;
+  const hasReachStep = (step: number) => {
+    switch (step) {
+      case 1:
+        return !hasOrganization;
+      case 2:
+        return hasOrganization && !hasWorkspace;
+      case 3:
+        return hasOrganization && hasWorkspace && !hasName;
+      case 4:
+        return hasOrganization && hasWorkspace && hasName;
+      default:
+        return false;
+    }
+  };
+
+  if (hasReachStep(ONBOARDING_STEPS.ORGANIZATION)) {
+    current = 1;
+  }
 
-  if (hasOrganization) {
+  if (hasReachStep(ONBOARDING_STEPS.WORKSPACE)) {
     current = 2;
   }
-  if (hasWorkspace) {
+
+  if (hasReachStep(ONBOARDING_STEPS.NAME)) {
     current = 3;
   }
-  if (hasName) {
+
+  if (hasReachStep(ONBOARDING_STEPS.PREFERENCE)) {
     current = 4;
   }
-  if (hasProfessionSegment) {
-    current = 5; // this step 5 doesnt exist so that will trigger the onboarding complete modal
-  }
-  if (!hasOrganization && !hasWorkspace && !hasName && !hasProfessionSegment) {
+
+  if (
+    !hasOrganization &&
+    !hasWorkspace &&
+    !hasName &&
+    (!hasProfessionSegment || !hasDeveloperSegment)
+  ) {
     current = 1;
   }
 
   return current;
 };
+
+export const getOnboardingStepsData = (member: Partial<MemberResponse>) => {
+  const { name, organization, workspace, profession_segment } = member;
```

Comment:
Should `developer_segment` be included here? Is it not included because it's not required unless the user selects `Work` from the `profession_segment` options in step 4?

Reply:
This should be removed, as it's not used anywhere. Thank you for catching that oversight.

Reply:
I ultimately used this function in my last commit to include the remaining form data in the payload request and incorporated the missing `developer_segment` that you caught. Thanks